### PR TITLE
fixes #72: A script teanslates English to other languages

### LINKED ISSUE #72: Create English to all other languages translation process
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/scribe-org/Scribe-Data/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow Scribe-Data's [Code of Conduct](https://github.com/scribe-org/Scribe-Data/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

The goal of this issue is to create a process whereby a single file is used to translate all words within []() to all other Scribe languages. To achieve this we'll be using [m2m100_418M](https://huggingface.co/facebook/m2m100_418M), with the output being a JSON file that has a string and keyed values for each language. This can then be transferred to an SQLite database table with each string in an index corresponding to a column value for each language.

Of specific importance is trying to get a metric of the accuracy of the translation and doing a cutoff such that we're no longer including low quality translations in Scribe applications :)

### Contribution

Happy to work on this as a first example of the translation process, and then others can take over for the other languages!

### PR DESCRIPTION
**Pull request by Linfye**
### Contributor checklist

- [] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

The script can run on Google Colab and that's where I code on. Because running the script will take too much time, the translated_words are only a part of all the words. But it shows the feasibility of the program.

Looking forward to your code review
### Related issue

- #ISSUE_NUMBER
#72

### CODE REVIEW

**Comment by andrewtavis** [comment]
Thanks for this, @Linfye! I'll get back to you with a review soon! Once this is merged you'd be welcome to work on the other languages :)

**Review by wkyoshida — COMMENTED** [review]
Awesome contribution @Linfye :rocket: Thanks so much for the work here!!

I'll let @andrewtavis add his review here as well, but just adding some comments of my own too with some ideas :wink:

File: src/scribe_data/extract_transform/languages/English/formatted_data/translated_words.json
Comment (andrewtavis) [reply src/scribe_data/extract_transform/languages/English/formatted_data/translated_words.json:845]: Happy to include it in here if @Linfye is comfortable with it :)

Comment (shashank-iitbhu) [reply src/scribe_data/extract_transform/languages/English/formatted_data/translated_words.json:845]: > This could make sense, but maybe we could do a separate issue for it?

@andrewtavis I already have a branch in my local where i have moved the tranlations functions to `utils.py`, If you allow i can open a separate PR for this. or can open a PR for #77 russian translation process.
then other contributors can take reference from that.

Comment (andrewtavis) [reply src/scribe_data/extract_transform/languages/English/formatted_data/translated_words.json:845]: A separate PR for this sounds good, @shashank-iitbhu! From there we can jump to the Russian :) Appreciate you reaching out here!

**Comment by Linfye** [comment]
I fixed the problems you mentioned except the last one. @wkyoshida I wonder now should I work on it or others works on it now. Looking forward to your reply.
cc @andrewtavis

**Comment by shashank-iitbhu** [comment]
> I fixed the problems you mentioned except the last one. @wkyoshida I wonder now should I work on it or others works on it now. Looking forward to your reply.
> cc @andrewtavis 

Can you refer to #88 and #89 ? I have implemented a different approach i.e batch processing of words for translation. This way it is relatively faster.

We can decide on a single approach, if the requirement is to iterate over each word rather than batch processing then we can go ahead with this PR.
cc @andrewtavis @wkyoshida

**Comment by Linfye** [comment]
> > I fixed the problems you mentioned except the last one. @wkyoshida I wonder now should I work on it or others works on it now. Looking forward to your reply.
> > cc @andrewtavis
> 
> Can you refer to #88 and #89 ? I have implemented a different approach i.e batch processing of words for translation. This way it is relatively faster.
> 
> We can decide on a single approach, if the requirement is to iterate over each word rather than batch processing then we can go ahead with this PR. cc @andrewtavis @wkyoshida

I check the code and wonder if continue downloading from last progress cause the words are too much. If it works better, we can adopt yours.

**Comment by andrewtavis** [comment]
Sorry for the delay on all of this, all :) I was on vacation and then sick right after... Checked and sent along some formatting in 2460584. I'll bring this in shortly as well as the work that's @shashank-iitbhu mentioned. I'll give it all a test to see how things are working. I'd say batch processing and having the process in the utils makes sense to me 😊

**Review by andrewtavis — APPROVED** [review]
Thanks for the great first contribution, @Linfye! Very important step in the the new translation process 😊

**Comment by andrewtavis** [comment]
Ah, and a quick note on this: let's be sure to remove as much whitespace from JSON outputs as possible in the future as that does bring the file size down slightly 😊

**Pull request merged**
